File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans = 0, n = nums.size(), currOne = 0, left = 0, deleted = 0;

        for (int right = 0; right < n; ++right){
            while (left < right && deleted > 1){
                if (nums[left++] == 0) --deleted;
                else --currOne;;
            }

            if (nums[right] == 0) ++deleted;
            else ++currOne;

            ans = max(ans, currOne);
        }

        if (ans > 0 && deleted == 0) --ans;

        return ans;
    }
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        deleted, left, n, ans = 0, 0, len(nums), 0
        curr_1 = 0

        for right in range(n):
            while left < right and deleted > 1:
                if nums[left] == 0:
                    deleted -= 1
                else:
                    curr_1 -= 1
                left += 1
            
            if nums[right] == 0:
                deleted += 1
            else:
                curr_1 += 1
            ans = max(ans, curr_1)

        if ans > 0 and deleted == 0:
            ans -= 1
        
        return ans